### PR TITLE
feat/nodejs-layer

### PR DESCRIPTION
**Pull request by samimusallam**

### CODE REVIEW

**Review by haddasbronfman — COMMENTED** [review]
Maybe we should add a README file.

File: nodejs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Logs
```
Comment (haddasbronfman) [inline nodejs/.gitignore]: My approach is to use minimal .gitignore file. The large files might be confusing and contain redundant info. (Of course it is your choice eventually :) )

File: nodejs/src/wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        await detectResources({
+            detectors: [awsLambdaDetector, envDetector, processDetector]
+        });
+
+    const provider = new NodeTracerProvider({
+        resource: detectedResources
+    });
+
+    const collectorOptions = {
+        url: 'https://production.cisco-udp.com/trace-collector',
```
Comment (osherv) [inline nodejs/src/wrapper.ts]: use cisco-specifications

File: nodejs/src/wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { detectResources, envDetector, processDetector } from '@opentelemetry/resources';
+import { getEnv } from '@opentelemetry/core';
+import { verifyToken } from './utils';
+
+
+const logLevel = getEnv().OTEL_LOG_LEVEL
+
+diag.setLogger(new DiagConsoleLogger(), logLevel)
+
+const ciscoToken = process.env["CISCO_TOKEN"] || '';
```
Comment (osherv) [inline nodejs/src/wrapper.ts]: use cisco-specifications

File: nodejs/src/wrapper.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { awsLambdaDetector } from '@opentelemetry/resource-detector-aws';
+import { detectResources, envDetector, processDetector } from '@opentelemetry/resources';
+import { getEnv } from '@opentelemetry/core';
+import { verifyToken } from './utils';
+
+
+const logLevel = getEnv().OTEL_LOG_LEVEL
```
Comment (osherv) [inline nodejs/src/wrapper.ts]: we would want cisco_debug i think.

File: nodejs/src/wrapper.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        },
+    })
+
+async function init() {
+    const detectedResources =
+        await detectResources({
+            detectors: [awsLambdaDetector, envDetector, processDetector]
+        });
+
+    const provider = new NodeTracerProvider({
```
Comment (osherv) [inline nodejs/src/wrapper.ts:32]: I think we will want to add Hexrays propogator as well.
deps:
 "@opentelemetry/id-generator-aws-xray": "^1.0.1",
    "@opentelemetry/sdk-trace-node": "^1.0.1",
    "@opentelemetry/propagator-aws-xray": "^1.0.1",
    "@opentelemetry/propagator-b3": "^1.0.1"

see here how to add propogators: 
https://github.com/aws-observability/aws-otel-lambda/blob/main/nodejs/wrapper-adot/src/adot-extension.ts

**Review by osherv — CHANGES_REQUESTED** [review]

File: nodejs/.gitignore (hunk 1 above)
Comment (samimusallam) [reply nodejs/.gitignore]: yea maybe this contains too many redundant things

**Review by osherv — APPROVED** [review]

**Pull request merged**